### ISSUE TITLE
Time for a "core team"?

### ISSUE THREAD

**Issue by mattetti** [mattetti/wd-sinatra#34]
Hey @kamui and @raul how do you guys feel if we setup an actual team and discuss this project together?

**Comment by raul** [comment]
Absolutely, I just landed but love the project's vision and would be glad to contribute and discuss its future.

**Comment by mattetti** [comment]
This project is many years old, I started it while at Sony and it's now being used at Sony and LivingSocial but I think it would greatly benefit from more input. The vision so far has been clear: allow developers to develop simple web APIs with a focus on 3 things:
- clearly defined and enforced interface
- simplicity
- performance

@kamui did an awesome job improving some parts of WD and moving into a modular approach so WD apps can run on their own or be mounted inside a Rails apps. This is the foundation of the 2.0 release but before we push this release, I think it would be nice to clean up the generated documentation as you (@raul) started doing and improve the WeaselDiesel API (support for custom headers, various responses, error codes etc..).

Someone reported a bug and I added the failing tests, but I haven't fixed the bug in question https://github.com/mattetti/Weasel-Diesel/commit/3ac386202ebc29cf703ddb0d35f2a68c28cf27f7  Now is probably a great time to define together what version 2.0 should be.

**Comment by raul** [comment]
Thanks for the nice introduction to the project Matt! I haven't used WD at work yet but I plan to use it for my side projects, and have a few suggestions and questions from what I've seen so far. I don't know how you guys prefer to discuss the project, so I'll drop them here for now:

#### Documentation

I agree with [this issue](https://github.com/mattetti/Weasel-Diesel/issues/15) and would like to move the doc. generation to WD (unless it's tied to wd_sinatra on purpose for a reason that I'm not aware of).

I also like [Matt's idea](https://github.com/mattetti/wd-sinatra/issues/12) of having an additional Markdown template which renders nicely in GitHub.

Large APIs may benefit from generating separate files linked from an index file – I haven't dealt with that scenario though, so I may be completely wrong. 

I would also like to improve the role of each endpoint's `overall` field (or an equivalent), I find it very important and would like to enforce its presence. The way I see it, documenting the purpose of an endpoint is as important as knowing its HTTP verb and URL, and is a better entry point to its documentation for human readers.

#### API

I wrote  a [small proposal](https://gist.github.com/raul/5652222) to enhance the responses' definitions by supporting different scenarios, headers and status codes. Before working on a first implementation I opted for working on the doc. generation because found it a good way to learn about the way WD works.

I like the response verification, I would make it optional on production and run it by default in development and test mode.

#### Organization

I've never designed a project like this so I'll take this as an opportunity to learn from you guys. When I took a look, I was a bit surprised by some design decisions:
- Doc being generated in wd_sinatra, already mentioned above.
- HTML being generated in WD (I tried to remove this dependency in my last PR).
- Sinatra and ActiveRecord are considered optional, but apps generated by wd_sinatra is opinionated enough to include helpers for configuring Airbrake and NewRelic (which are great services). Should we extract those as pluggable side projects?

#### Team

Do you guys like us to sync/discuss via GH issues or perhaps prefer a mailing list? I don't have any strong preference.

#### Next steps

As time permits, I can work on fixing the bug that Matt mentioned and then take a look to the rest while we discuss the project roadmap. Exciting times ahead! :)

**Comment by kamui** [comment]
I'd definitely be interested as well. Sorry I can't give a more detailed response at the moment, but I will when I get a chance!

**Comment by mattetti** [comment]
> Do you guys like us to sync/discuss via GH issues or perhaps prefer a mailing list? I don't have any strong preference.

I don't have any preferences either, whatever works better for you guys. I'm also on #shitfire (freenode) most of the time (though I'm currently traveling).

**Comment by kamui** [comment]
I'd be fine with GH, but I think some discussions don't fit into the concept of a ticket/issue. I think we probably should have a Google Group too, partly for users to ask questions and partly for longer form discussions that don't quite fit into the GH format.

**Comment by mattetti** [comment]
Created a group, sending you an invite

**Comment by kamui** [comment]
Just putting the link here: https://groups.google.com/group/weaseldiesel
